server/auth/providers/gitea.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import Router from "koa-router";
+//import { capitalize } from "lodash";
+import { Strategy as GiteaStrategy } from "passport-gitea";
+import accountProvisioner from "../../commands/accountProvisioner";
+import env from "../../env";
+import {
+  GiteaOrganizationError,
+  GiteaOrganizationAPIError,
+} from "../../errors";
+import passportMiddleware from "../../middlewares/passport";
+//import { getAllowedDomains } from "../../utils/authentication";
+import { StateStore } from "../../utils/passport";
+
+const router = new Router();
+const providerName = "gitea";
+const GITEA_CLIENT_ID = process.env.GITEA_CLIENT_ID;
+const GITEA_CLIENT_SECRET = process.env.GITEA_CLIENT_SECRET;
+const GITEA_URL = process.env.GITEA_URL; //Usage of custom url
```

Comment:
Should this default to the main service so that it can be completely unset in that case?